pandas-stubs/_typing.pyi

```diff
@@ -738,7 +739,20 @@ JsonSeriesOrient: TypeAlias = Literal["split", "records", "index", "table"]
 TimestampConvention: TypeAlias = Literal["start", "end", "s", "e"]
 
 CSVEngine: TypeAlias = Literal["c", "python", "pyarrow", "python-fwf"]
-CSVQuoting: TypeAlias = Literal[0, 1, 2, 3]
+# [pandas-dev/pandas-stubs/991]
+# Ref: https://github.com/python/cpython/blob/5a4fb7ea1c96f67dbb3df5d4ccaf3f66a1e19731/Modules/_csv.c#L88-L91
+# QUOTE_MINIMAL = 0
+# QUOTE_ALL = 1
+# QUOTE_NONNUMERIC = 2
+# QUOTE_NONE = 3
+# Added in 3.12:
+# QUOTE_STRINGS = 4
+# QUOTE_NOTNULL = 5
+CSVQuotingCompat: TypeAlias = Literal[0, 1, 2, 3]
+if sys.version_info < (3, 12):
+    CSVQuoting: TypeAlias = CSVQuotingCompat
+else:
+    CSVQuoting: TypeAlias = CSVQuotingCompat | Literal[4, 5]
```

Comment:
Can you try the `Union` idea here, i.e., use `Union[csv.QUOTE_MINIMAL, # other values]` instead of `Literal`

Reply:
No go.
Both type checker can't resolve the type of the variable at static time, might be an edge case since the Enums are actually defined in the C compiled code, or they might be extra strict with variable values
![image](https://github.com/user-attachments/assets/936478cc-dd64-4112-8c9b-043fb196667f)